File: src/lessons/welcome/array/arraycount9/ArrayCount9Entity.java
package lessons.welcome.array.arraycount9;

import static plm.core.ValueSerializer.*;

import plm.universe.bat.BatEntity;

public class ArrayCount9Entity extends BatEntity {

  public void run()
  {
    int count = getTestCount();
    for (int i = 0; i < count; i++) {
      Object[] param = (Object[])deserialize(getTest(i));
      setTestResult(i, serialize(arrayCount9((int[])param[0])));
    }
  }

  /* BEGIN TEMPLATE */
  int arrayCount9(int[] nums)
  {
    /* BEGIN SOLUTION */
    int count = 0;
    for (int i = 0; i < nums.length; i++) {
      if (nums[i] == 9) {
        count++;
      }
    }
    return count;
    /* END SOLUTION */
  }
  /* END TEMPLATE */
}

File: src/lessons/welcome/array/arraycount9/ArrayCount9Entity.scala
package lessons.welcome.array.arraycount9

import plm.universe.bat.BatEntity
import plm.universe.bat.BatTest

class ArrayCount9Entity extends BatEntity {

    override def run() {
      import plm.core.ValueSerializer._

      val count = getTestCount()
      for (i <- 0 to count -1) {
        val param = deserialize(getTest(i)).asInstanceOf[Array[Object]]
      	setTestResult(i, serialize(arrayCount9(param(0).asInstanceOf[Array[Int]]) ))
	  }
	}

	/* BEGIN TEMPLATE */
	def arrayCount9(nums:Array[Int]): Int = {
		/* BEGIN SOLUTION */
	  var res = 0
	  for (value <- nums)
	    if (value == 9)
	      res += 1
	  return res
		/* END SOLUTION */
	}
	/* END TEMPLATE */
}
